src/jdk.jshell/share/classes/jdk/jshell/execution/LocalExecutionControl.java

```diff
@@ -133,7 +194,11 @@ public void stop() throws EngineTerminationException, InternalException {
             if (execThreadGroup == null) {
                 throw new InternalException("Process-local code snippets thread group is null. Aborting stop.");
             }
-
+            try {
+                allStop.set(null, true);
+            } catch (IllegalArgumentException| IllegalAccessException ex) {
+                throw new InternalException(ex.getMessage());
```

Comment:
to avoid to drop the stack trace of the original exception, it's better to chain the exceptions,
```
  throw (InternalException) new InternalException().initCause(ex);
```

Reply:
I just followed similar pattern as in other 9 cases of throwing new InternalException in idk.jshell.

Reply:
Adjusted exception message construction to exactly match other cases.